hs-bindgen/src-internal/HsBindgen/Frontend/Pass/Select/IsPass.hs

```diff
@@ -93,7 +93,7 @@ data SelectConfig = SelectConfig {
 -- | Select trace messages
 data SelectMsg =
     SelectTransitiveDependencyUnavailable C.NsPrelimDeclId
-  | SelectExcluded (C.DeclInfo Select)
+  | SelectNotSelected (C.DeclInfo Select)
```

Comment:
Report reason of why something was selected (either by predicate, or because it is a transitive dependency).

Reply:
If easy, add the reason why it is a transitive case.

Reply:
Reporting the root declaration causing the transitive dependency is hard, because we unified the resolution of transitive dependencies into one function call (traversing all roots in one go). We had the resolution separate (i.e., per declaration root) in the beginning but decided to optimize the algorithm.